Classes/common/HTTP/CDTURLSession.m

```diff
@@ -53,8 +56,28 @@ - (NSURLSessionDataTask *)dataTaskWithRequest:(NSURLRequest *)request
                                                         NSURLResponse *response,
                                                         NSError *error))completionHandler
 {
+    
+    //make the context
+    CDTURLSessionInterceptorContext *ctx = [[CDTURLSessionInterceptorContext alloc ]initWithRequest:[request mutableCopy]];
+    for ( id obj in self.requestInterceptors) {
+        
+        if( [obj conformsToProtocol:@protocol(CDTURLSessionInterceptor)]){
+            id<CDTURLSessionInterceptor> interceptor = obj;
+            
+            if([interceptor respondsToSelector:@selector(interceptRequestInContext:)]){
+                ctx = [interceptor interceptRequestInContext:ctx];
+            } else {
+                CDTLogWarn(CDTREPLICATION_LOG_CONTEXT, @"%@ doesnn't respond to \
+                           interceptRequestInContext: skipping",interceptor);
+            }
+        } else {
+            CDTLogWarn(CDTREPLICATION_LOG_CONTEXT, @"%@ doesn't conform to protocol \
+                       CDTURLSessionInterceptor, skipping",obj);
+        }
+    }
+    
     __weak CDTURLSession *weakSelf = self;
-    NSURLSessionDataTask *task = [self.session dataTaskWithRequest:request
+    NSURLSessionDataTask *task = [self.session dataTaskWithRequest:[ctx.request copy]
```

Comment:
[Apple's documentation](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Protocols/NSCopying_Protocol/index.html#//apple_ref/occ/intf/NSCopying) doesn't provide information either way, but I would interpret `a copy must be a functionally independent object with values identical to the original at the time the copy was made` as it being a deep copy.